packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -317,9 +322,11 @@ public class InAppPurchasePlugin: NSObject, FlutterPlugin, InAppPurchaseAPI {
         guard let messenger = registrar?.messenger() else {
           fatalError("registrar.messenger can not be nil.")
         }
-        paymentQueueDelegateCallbackChannel = FlutterMethodChannel(
-          name: "plugins.flutter.io/in_app_purchase_payment_queue_delegate",
-          binaryMessenger: messenger)
+        paymentQueueDelegateCallbackChannel = DefaultMethodChannel(
```

Comment:
why is the implementation relying on DefaultMethodChannel rather than MethodChannel protocol?

Reply:
wait actually its not
self.paymentQueueDelegateCallbackChannel has type MethodChannel protocol

Reply:
Sorry i intended to type "you *can't* hard code using `DefaultMethodChannel` here. ". If you need this channel to be swapped during test, you must not directly depend on `DefaultMethodChannel`. You can instead inject a factory block that returns the protocol type (rather than the concrete type)

Reply:
Wait im not understanding this part "If you need this channel to be swapped during test, you must not directly depend on DefaultMethodChannel". `transactionObserverCallbackChannel` here
https://github.com/flutter/packages/blob/4cd2042a2a4df792ca1845891b9afb6b69cb233e/packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift#L95
Is declared in the same way, but I can test it just fine like here -

https://github.com/flutter/packages/blob/4cd2042a2a4df792ca1845891b9afb6b69cb233e/packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m#L766C1-L766C13

Why is this different? Why can't I just be like `self.paymentQueueDelegateCallbackChannel = testChannel`

Reply:
ic. if it's the existing behavior you can keep it for now. 

It's a bad practice to expose an internal property and overwrite it during test. `paymentQueueDelegateCallbackChannel` shouldn't have been exposed as part of the API in the first place.

Reply:
I don't find `self.paymentQueueDelegateCallbackChannel = testChannel` in the test. Is this function tested?

Reply:
the method channel being called with the correct params is tested in FIAPPaymentQueueDelegateTests.m here
https://github.com/flutter/packages/pull/6862/files#diff-9576fa01a58efbb6f16cabaa39e0036da964878c1c8b018d3379a7f6cd2b1bacR49

Reply:
The mocked channel is just directly being set when initing FIAPPaymentQueueDelegate in that test ^
But the registerPaymentQueueDelegateWithError function is being tested  here https://github.com/flutter/packages/pull/6862/files#diff-fafa156a83a9664b053836950cb331366572ef247344b5071ba5f23316965afcR683

Reply:
The only reason why i think paymentQueueDelegateCallbackChannel is stored in a privare var is so it can get set to nil in removePaymentQueueDelegateWithError

Reply:
okay cool, changed it back to FlutterMethodChannel